ethereum/contracts/relayer/coreRelayer/CoreRelayerDelivery.sol

```diff
@@ -1,623 +1,560 @@
+
+    unchecked{sendParams.maxTransactionFee += increaseAmount;}
+    unchecked{remainingRefundAmount = maxIncreaseAmount - increaseAmount;}
+  }
+
+  function payRefunds(
+    DeliveryInstruction memory deliveryInstruction,
+    address payable relayerRefundAddress,
+    uint256 transactionFeeRefundAmount,
+    bool receiverValueWasPaid,
+    uint256 transactionFeeRefundAmountPostForward
+  ) private returns (RefundStatus refundStatus) {
+    //Amount of receiverValue that is refunded to the user (0 if the call to
+    //  'receiveWormholeMessages' did not revert, or the full receiverValue otherwise)
+    uint256 receiverValueRefundAmount =
+      (receiverValueWasPaid ? 0 : deliveryInstruction.receiverValueTarget);
+
+    //Total refund to the user
+    uint256 refundToRefundAddress =
+      receiverValueRefundAmount + transactionFeeRefundAmountPostForward;
+    
+    //Refund the user
+    refundStatus = payRefundToRefundAddress(
+      deliveryInstruction.refundChainId,
+      deliveryInstruction.refundAddress,
+      refundToRefundAddress,
+      deliveryInstruction.targetRelayProvider
+    );
 
-    function processOverrides(
-        IWormholeRelayerInternalStructs.DeliveryInstruction memory deliveryInstruction,
-        bytes memory encoded
-    )
-        internal
-        pure
-        returns (
-            IWormholeRelayerInternalStructs.DeliveryInstruction memory withOverrides,
-            bytes32 redelivery
-        )
-    {
-        if (encoded.length == 0) {
-            return (deliveryInstruction, 0x0);
-        } else {
-            IDelivery.DeliveryOverride memory overrides = decodeDeliveryOverride(encoded);
-            if (overrides.gasLimit < deliveryInstruction.executionParameters.gasLimit) {
-                revert IDelivery.InvalidOverrideGasLimit();
-            } else if (overrides.receiverValue < deliveryInstruction.receiverValueTarget) {
-                revert IDelivery.InvalidOverrideReceiverValue();
-            } else if (overrides.maximumRefund < deliveryInstruction.maximumRefundTarget) {
-                revert IDelivery.InvalidOverrideMaximumRefund();
-            }
-
-            deliveryInstruction.executionParameters.gasLimit = overrides.gasLimit;
-            deliveryInstruction.receiverValueTarget = overrides.receiverValue;
-            deliveryInstruction.maximumRefundTarget = overrides.maximumRefund;
-            return (deliveryInstruction, overrides.redeliveryHash);
-        }
+    //Refund the relayer (their extra funds) + (the amount that the relayer spent on gas)
+    //  + (the users refund if that refund didn't succeed)
+    uint256 relayerRefundAmount = (
+      msg.value - deliveryInstruction.receiverValueTarget - deliveryInstruction.maximumRefundTarget
+    ) + (deliveryInstruction.maximumRefundTarget - transactionFeeRefundAmount)
+    //TODO AMO: Isn't this a bug? We add the same amount regardless of whether we hit the max or not
```

Comment:
- Don’t believe this is a bug, but definitely an interesting quirk: for cross chain refunds, if the refund amount ends up being more than the maximum budget, the relay provider gets paid the full amount and only has to deliver the ‘maximum budget’ amount on the refund chain. This was an intentional decision and I think it is fine? because, 1) I assume the maximum budget will be quite large 2) we can make the guarantee: you will receive refunds up to the maximum budget if you are doing a cross-chain refund; I think that is reasonable, and 3) we can consider the price graph not linear but piecewise linear and then straight vertical, in which case this is correct. Lmk if I am missing something about your comment

Reply:
No, sounds good! Definitely deserves a comment in the code though, I think. If you truncate the funds, it's a very natural to ask "but what about the rest?" and so there should be a comment there to answer that. (Or perhaps just point to the part of the bigger doc that explains it)